metals/src/main/scala/scala/meta/internal/parsing/BloopDiagnosticsParser.scala

```diff
@@ -0,0 +1,73 @@
+package scala.meta.internal.parsing
+
+import java.nio.file.Paths
+
+import scala.jdk.CollectionConverters.SeqHasAsJava
+
+import org.eclipse.lsp4j.Diagnostic
+import org.eclipse.lsp4j.DiagnosticSeverity
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.PublishDiagnosticsParams
+import org.eclipse.lsp4j.Range
+
+case class FileDiagnostic(filepath: String, diagnostic: Diagnostic)
+
+object BloopDiagnosticsParser {
+  def getDiagnosticsFromErrors(
+      errorLines: Array[String]
+  ): Iterable[PublishDiagnosticsParams] = {
+    errorLines
+      .foldLeft(List.empty[Vector[String]]) { case (accumulator, element) =>
+        if (element.startsWith("/")) {
```

Comment:
Maybe add some comments on specific conditions that might be unclear at first glance?